### PR TITLE
Add vnc_keymap attribute

### PR DESCRIPTION
**Pull request by rsalevsky**

### CODE REVIEW

File: chef/data_bags/crowbar/migrate/nova/008_add_vnc_keymap.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+def upgrade ta, td, a, d
```
Comment (dirkmueller) [inline chef/data_bags/crowbar/migrate/nova/008_add_vnc_keymap.rb]: File needs to be named with 021_....

**Comment by vuntz** [comment]
I'm thinking it'd make sense to expose this setting in the crowbar UI.

File: chef/data_bags/crowbar/bc-template-nova.schema [hunk 2]
```diff
@@ -39,6 +39,7 @@
             "pfs_deps": { "type": "seq", "required": true, "sequence": [ { "type": "str" } ] },
             "neutron_metadata_proxy_shared_secret": { "type": "str" },
             "neutron_url_timeout": {"type": "int"},
+            "vnc_keymap": { "type": "str" , "required": true},
```
Comment (dirkmueller) [inline chef/data_bags/crowbar/bc-template-nova.schema]: missing space between "true" and "}"

**Comment by dirkmueller** [comment]
@vuntz yeah, but a free text field is not so sexy. and a combobox is hard to fill with valid entries I guess.. :/

**Comment by tboerger** [comment]
I'm fine with a text input as long as we don't have a list of valid values.

**Comment by tboerger** [comment]
:+1:

File: crowbar_framework/app/views/barclamp/nova/_edit_attributes.html.haml [hunk 3]
```diff
@@ -15,6 +15,7 @@
       = boolean_field :trusted_flavors
 
     = boolean_field :use_novnc
+    = string_field :vnc_keymap
```
Comment (vuntz) [inline crowbar_framework/app/views/barclamp/nova/_edit_attributes.html.haml]: Would it make sense to create a `%fieldset` for VNC settings?

**Comment by vuntz** [comment]
+1

**Comment by vuntz** [comment]
Actually: are you able to make this option work? I do see the keymap correctly set in libvirt, but in vnc, it's still the american keymap :/

**Comment by rsalevsky** [comment]
Yes, when I use the German special characters on en-us keymap then I get `atkbd serio0: Use 'setkeycodes 00 <keycode>' to make it known.` But when I change the keymap to de then it works.

**Comment by dirkmueller** [comment]
LGTM, +1

**Pull request merged**